### ISSUE TITLE
Possible to refactor date/reminder functionality?

### ISSUE THREAD

**Issue by geckolinux** [spacecowboy/NotePad#285]
Hi there, thanks a lot for creating this extremely useful and clean app. I wonder if it it would be possible to re-work the date and reminder functionality a bit? In my use case, most of my todo items just have a due date, with no time. But when I do set a time, I usually don't need a reminder, as I frequently check my list. But then again, there are a few extremely important or time-sensitive issues where I do need a reminder. So I would prefer an option for each task to have a picker "Due date" and "Due time", both being optional. Then, when adding a reminder, maybe it could use relative times by default, such as "1 day before the event" or "30 minutes before the event" or "Right at the due date and time". It would still be useful to be able to add additional reminders for specific dates and times too.

Additionally, I would really like to see issues #281 and #251 added, but with a tweak: Instead of recurring reminders, would it be possible to have recurring due dates? So when I check the item as completed, it could re-schedule the due date to the next nth X. And if the previous tweak regarding relative reminder dates/times was implemented, these would come along for free every time relative to the new Due date.

Thanks a lot for your time and hard work!

**Comment by spacecowboy** [comment]
Thanks for the feedback. Your thought are very much aligned with how I have imagined a possible refactoring. The only problem is that such a refactoring would make the app even more incompatible with Google Tasks.

Personally, I'd like to axe Google Tasks support entirely and focus on file storage (SD-card/Dropbox/Google Drive). But there are a lot of users (I think) who find the Google Tasks support essential, so I'm not gonna remove it.

Making a second app is problematic, since then my already limited time would be split between more projects, which has its obvious flaws.

So TLDR, I want to do it I just don't have the time :/

**Comment by sb56637** [comment]
Ah, I understand. I would personally also like a simple SD and/or Google Drive backend. But I appreciate your time, and I'm sure all your users appreciate the time you dedicate to this app.

Cheers!

**Comment by spacecowboy** [comment]
You already have an SD Card backend though :)

**Comment by CampelloManuel** [comment]
will work on this in #251
